cmd/soroban-rpc/internal/methods/simulate_transaction.go

```diff
@@ -70,13 +72,34 @@ func NewSimulateTransactionHandler(logger *log.Entry, getter PreflightGetter) jr
 			}
 		}
 
-		result, err := getter.GetPreflight(ctx, sourceAccount, xdrOp)
+		jobBufferLen, jobBufferCapacity := pwp.GetJobQueueLenAndCapacity()
+		if jobBufferLen != 0 && jobBufferLen == jobBufferCapacity {
+			return SimulateTransactionResponse{
+				Error: "All workers are busy, try again later",
+			}
+		}
```

Comment:
We discussed returning a 429 HTTP error here but, unfortunately that's not really how JSON RPC works but:

1. I don't think that's how you are supposed to use JSON RPC (you should have a response object for every request). Note that you get a stream of JSON request objects and the client expects a matching stream of responses. See https://www.jsonrpc.org/specification .
2. The Go library we are using doesn't allow overriding the HTTP error we are getting.

We may want to reserve and return one of the `-32000 to -32099` server errors for this though.

I am aware this will make it more difficult for loadbalancers to detect and I am open to alternatives (like having a global approach for all endpoints).